homeassistant/components/octoprint/__init__.py

```diff
@@ -215,7 +221,9 @@ def get(self, endpoint):
 
         url = self.api_url + endpoint
         try:
-            response = requests.get(url, headers=self.headers, timeout=9)
+            response = requests.get(
+                url, headers=self.headers, timeout=9, verify=self.verify_ssl
```

Comment:
We shouldn't extend this code as we don't want it in home assistant at all. It should be moved to a standalone library published on pypi.

https://developers.home-assistant.io/docs/en/creating_component_code_review.html#4-communication-with-devicesservices

Reply:
It's not clear to me how `headers=self.headers` is acceptable but `verify=self.very_ssl` isn't acceptable.  🤷‍♂️
It's extending on Requests, which a quick look at [Requests2](https://pypi.org/project/requests2/) (currently used by HA) shows this as available since 2011-12-28.

Could you clarify?

Reply:
We don't want any device specific I/O code at all inside home assistant. The existing code is legacy before we had this requirement. All the requests related code and data parsing should move to a separate pypi library.

Reply:
Thanks for the distinction @MartinHjelmare.
I understand that is legacy code now. I'll have an attempt at publishing my changes via standalone pypi library.